UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/MetaTileMap.cs

```diff
@@ -1,587 +1,1316 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
 using UnityEngine;
 using UnityEngine.Events;
+using Debug = UnityEngine.Debug;
 
-[ExecuteInEditMode]
-public class MetaTileMap : MonoBehaviour
+namespace TileManagement
 {
-	/// <summary>
-	/// Use this dictionary only if performance isn't critical, otherwise try using arrays below
-	/// </summary>
-	public Dictionary<LayerType, Layer> Layers { get; private set; }
-
-	//Using arrays for iteration speed
-	public LayerType[] LayersKeys { get; private set; }
-	public Layer[] LayersValues { get; private set; }
-	public ObjectLayer ObjectLayer { get; private set; }
-
-
-	public List<Layer> ffLayersValues;
-	/// <summary>
-	/// Array of only layers that can ever contain solid stuff
-	/// </summary>
-	public Layer[] SolidLayersValues { get; private set; }
-	/// <summary>
-	/// Layers that contain TilemapDamage
-	/// </summary>
-	public Layer[] DamageableLayers { get; private set; }
-
-	public float Resistance(Vector3Int cellPos, bool includeObjects = true)
+	[ExecuteInEditMode]
+	public class MetaTileMap : MonoBehaviour
 	{
-		float resistance = 0;
-		foreach ( var damageableLayer in DamageableLayers )
-		{
-			resistance += damageableLayer.TilemapDamage.Integrity( cellPos );
-		}
+		public int TargetMSpreFrame = 5;
 
-		if ( includeObjects && ObjectLayer )
-		{
-			resistance += ObjectLayer.GetObjectResistanceAt( cellPos, true );
-		}
-		return resistance;
-	}
+		public Stopwatch stopwatch = new Stopwatch();
 
-	private void OnEnable()
-	{
-		Layers = new Dictionary<LayerType, Layer>();
-		var layersKeys = new List<LayerType>();
-		var layersValues = new List<Layer>();
-		var solidLayersValues = new List<Layer>();
-		var damageableLayersValues = new List<Layer>();
 
-		foreach (Layer layer in GetComponentsInChildren<Layer>(true))
+		private Dictionary<Layer, Dictionary<Vector3Int, TileLocation>> PresentTiles =
+			new Dictionary<Layer, Dictionary<Vector3Int, TileLocation>>();
+
+		/// <summary>
+		/// Use this dictionary only if performance isn't critical, otherwise try using arrays below
+		/// </summary>
+		public Dictionary<LayerType, Layer> Layers { get; private set; }
+
+		public static Stack<TileLocation> PooledTileLocation = new Stack<TileLocation>();
```

Comment:
Codacy found an issue: [Change the visibility of 'PooledTileLocation' or make it 'const' or 'readonly'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6319358)